src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,594 @@
+
+            var tcs = new TaskCompletionSource<HttpResponseMessage>();
+
+            // Wrap the cancellationToken in a using so that it can be disposed of whether
+            // we successfully fetched or failed trying.
+            using (cancellationToken.Register(() => tcs.TrySetCanceled()))
+            {
+                try
+                {
+                    var requestObject = new JSObject();
+
+                    if (request.Properties.TryGetValue("WebAssemblyFetchOptions", out object? fetchOoptionsValue) &&
+                        fetchOoptionsValue is IDictionary<string, object> fetchOptions)
+                    {
+                        foreach (KeyValuePair<string, object> item in fetchOptions)
+                        {
+                            requestObject.SetObjectProperty(item.Key, item.Value);
+                        }
+                    }
+
+                    requestObject.SetObjectProperty("method", request.Method.Method);
+
+                    // We need to check for body content
+                    if (request.Content != null)
+                    {
+                        if (request.Content is StringContent)
+                        {
+                            requestObject.SetObjectProperty("body", await request.Content.ReadAsStringAsync().ConfigureAwait(continueOnCapturedContext: true));
+                        }
+                        else
+                        {
+                            // 2.1.801 seems to have a problem with the line
+                            // using (var uint8Buffer = Uint8Array.From(await request.Content.ReadAsByteArrayAsync ()))
+                            // so we split it up into two lines.
+                            var byteAsync = await request.Content.ReadAsByteArrayAsync().ConfigureAwait(continueOnCapturedContext: true);
+                            using (Uint8Array uint8Buffer = Uint8Array.From(byteAsync))
+                            {
+                                requestObject.SetObjectProperty("body", uint8Buffer);
+                            }
+                        }
+                    }
+
+                    // Process headers
+                    // Cors has it's own restrictions on headers.
+                    // https://developer.mozilla.org/en-US/docs/Web/API/Headers
+                    using (HostObject jsHeaders = new HostObject("Headers"))
+                    {
+                        if (request.Headers != null)
+                        {
+                            foreach (KeyValuePair<string, IEnumerable<string>> header in request.Headers)
+                            {
+                                foreach (string value in header.Value)
+                                {
+                                    jsHeaders.Invoke("append", header.Key, value);
+                                }
+                            }
+                        }
+                        if (request.Content?.Headers != null)
```

Comment:
Headers will never be null (Content might be null):
```suggestion
                        if (request.Content != null)
```